### PR TITLE
Edit section modal

### PR DESCRIPTION
**Pull request by jessywork**

### CODE REVIEW

File: GrocifyApp.UI/src/components/modals/BaseModal.jsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const BaseModal = ({ isOpen, onClose, titleModal, modalBody, onConfirm, isButton
             
                     <div className={`${styles.modalMobile + " modal-mobile"} ${isOpen ? styles.open : ''} ${noFooter ? styles.noFooter : ''}`}>
                         <div className={styles.modalHeader + " modal-header"}>
-                            <div className={"title title--s weight--m " + styles.title}>{titleModal}</div>
+                            <div className={"title title--s weight--m " + styles.title} dangerouslySetInnerHTML={{ __html: titleModal }}/>
```
Comment (Catarina98) [inline GrocifyApp.UI/src/components/modals/BaseModal.jsx:29]: what is this?

File: GrocifyApp.UI/src/pages/Products/ProductSections.jsx [hunk 2]
```diff
@@ -90,16 +89,31 @@ function ProductSections() {
...
+        setIsModalOpen(true);
+    };
+
     return (
         <Layout>
 
             {selectedSection != null && isModalDeleteOpen && (
                 <BaseModal isConfirmModal={true} isOpen={isModalDeleteOpen} onClose={() => closeDeleteModal()} onConfirm={deleteSection}
-                    titleModal={ModalConsts.DeleteTitle(`"${selectedSection.name}" section`)} />)}
+                    titleModal={ModalConsts.DeleteTitle(`<span class="color--primary">${selectedSection.name}</span> ` + GenericConsts.Section)} />)}
```
Comment (Catarina98) [inline GrocifyApp.UI/src/pages/Products/ProductSections.jsx:106]: align

File: GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import ApiEndpoints from '../../../consts/ApiEndpoints';
+
+const ProductSectionModal = ({ onClose, onConfirm, section }) => {
+    const [isButtonDisabled, setButtonDisabled] = useState(true);
+    const [productSectionName, setProductSectionName] = useState(section != null ? section.name : '');
+    const [productSectionIcon, setProductSectionIcon] = useState(section != null ? section.icon : 'Home');
+    const { makeRequest } = useApiRequest();
+
+    useEffect(() => {
+        if (section) {
```
Comment (Catarina98) [inline GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx]: please review, when editting, I'm able to remove section name and save it

File: GrocifyApp.UI/src/components/modals/BaseModal.jsx (hunk 1 above)
Comment (jessywork) [reply GrocifyApp.UI/src/components/modals/BaseModal.jsx:29]: is to render html tags on consts

File: GrocifyApp.UI/src/pages/Products/ProductSections.jsx (hunk 2 above)
Comment (jessywork) [reply GrocifyApp.UI/src/pages/Products/ProductSections.jsx:106]: ignore git. this is align on IDE

File: GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx (hunk 3 above)
Comment (Catarina98) [reply GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx]: consider the following to be more easy to read:

`
const isSectionEdited = section && (productSectionName !== section.name || productSectionIcon !== section.icon);
const areInputsValid = productSectionName !== "" && productSectionIcon !== "";
        
setButtonDisabled(!areInputsValid || (section && !isSectionEdited));
`

File: GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+//Assets & Css
+import styles from './ProductSectionModal.module.scss';
+
+//Consts
+import { PlaceholderConsts, LabelConsts, ButtonConsts, ModalConsts } from '../../../consts/ENConsts';
+import InputType from '../../../consts/InputType';
+import ApiEndpoints from '../../../consts/ApiEndpoints';
+
+const ProductSectionModal = ({ onClose, onConfirm, section }) => {
```
Comment (Catarina98) [inline GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx]: rename section to sectionToUpdate

**Review by Catarina98 — APPROVED** [review]

**Pull request merged**